src/parser/function.js

```diff
@@ -266,14 +266,27 @@ module.exports = {
   },
   /**
    * ```ebnf
-   *    argument_list ::= T_ELLIPSIS? expr
+   *    argument_list ::= T_STRING ':' expr | T_ELLIPSIS? expr
    * ```
    */
   read_argument: function () {
     if (this.token === this.tok.T_ELLIPSIS) {
       return this.node("variadic")(this.next().read_expr());
     }
-    return this.read_expr();
+    const name = this.text();
+    let res;
+    try {
+      res = this.read_expr();
```

Comment:
hi, not sure that's the right way, you can also go ahead scan next tokens to see f it matches a T_STRING  + ':' and if not go for expr call

Reply:
fixed :slightly_smiling_face: